docs/notebooks/multi_objective_ehvi.pct.py

```diff
@@ -0,0 +1,219 @@
+    grid_density=100,
+    contour=True,
+    title=["Obj 1", "Obj 2"],
+    figsize=(12, 6),
+    colorbar=True,
+    xlabel="$X_1$",
+    ylabel="$X_2$",
+)
+plot_bo_points(initial_query_points, ax=ax[0, 0], num_init=num_initial_points)
+plot_bo_points(initial_query_points, ax=ax[0, 1], num_init=num_initial_points)
+plt.show()
+
+# %% [markdown]
+# ... and in the objective space
+
+# %%
+plot_mobo_points_in_obj_space(initial_data[OBJECTIVE].observations)
+plt.show()
+
+
+# %% [markdown]
+# ## Modelling the two functions
+#
+# In this example we model the two objective functions individually with their own Gaussian process models. For problems where the objective functions are similar it may make sense to build a joint model rather than a model stack.
+
+
+# %%
+def create_single_outcome_model(data, outcome_idx, input_dim=2, l=1.0):
+    variance = tf.math.reduce_variance(tf.gather(data.observations, [outcome_idx], axis=1))
+    lengthscale = l * tf.ones(input_dim, dtype=gpflow.default_float())
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=lengthscale)
+    single_outcome_dataset = Dataset(
+        data.query_points, tf.gather(data.observations, [outcome_idx], axis=1)
+    )
+    gpr = gpflow.models.GPR(single_outcome_dataset.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+    return create_model(
+        {
+            "model": gpr,
+            "optimizer": gpflow.optimizers.Scipy(),
+            "optimizer_args": {
+                "minimize_args": {"options": dict(maxiter=100)},
+            },
+        }
+    )
+
+
+# %%
+objective_models = [
+    (
+        create_single_outcome_model(
+            initial_data[OBJECTIVE],
+            outcome_idx=i,
+        ),
+        1,
+    )
+    for i in range(num_objective)
+]
+
+# %% [markdown]
+# Stack the two independent GP model in a single `ModelStack` multi-output model.
+
+# %%
+models = {OBJECTIVE: ModelStack(*objective_models)}
+
+# %% [markdown]
+# ## Define the acquisition function
+# Here we utilize the [EHVI](https://link.springer.com/article/10.1007/s10898-019-00798-7): `ExpectedHypervolumeImprovement` acquisition function:
+
+# %%
+ehvi = ExpectedHypervolumeImprovement()
+rule: EfficientGlobalOptimization[Box] = EfficientGlobalOptimization(builder=ehvi.using(OBJECTIVE))
+
+# %% [markdown]
+# ## Run the optimization loop
+#
+# We can now run the optimization loop
+
+# %%
+num_steps = 20
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+result = bo.optimize(num_steps, initial_data, models, acquisition_rule=rule)
+
+# %% [markdown]
+# To conclude, we visualize the queried data across the design space.  
+# The cross represented dots are initial sampled points by `search_space.sample`, the dots are bo-obtained points.
+
+# %%
+datasets = result.try_get_final_datasets()
+data_query_points = datasets[OBJECTIVE].query_points.numpy()
+data_observations = datasets[OBJECTIVE].observations.numpy()
+
```

Comment:
Why have we gone to numpy here?

Reply:
The earlier reason is thinking as the `plotting.py` module is numpy based, there might has an expectation of transforming input to numpy (like [ei notebook](https://github.com/secondmind-labs/trieste/blob/b75db97734f6b9ea6ffa04f720ec178e4764fe35/docs/notebooks/expected_improvement.pct.py#L94) does), but am actually not sure the code design strategy of this module and how we should really handle it.

Reply:
Removed.